apps/web/pages/event-types/[type].tsx

```diff
@@ -92,6 +102,366 @@ const addDefaultLocationOptions = (
   });
 };
 
+type TWebhook = inferQueryOutput<"viewer.webhook.list">[number];
+
+function WebhookListItem(props: { webhook: TWebhook; onEditWebhook: () => void }) {
+  const { t } = useLocale();
+  const utils = trpc.useContext();
+  const deleteWebhook = trpc.useMutation("viewer.webhook.delete", {
+    async onSuccess() {
+      await utils.invalidateQueries(["viewer.webhook.list"]);
+    },
+  });
+
+  return (
+    <ListItem className="-mt-px flex w-full p-4">
+      <div className="flex w-full justify-between">
+        <div className="flex max-w-full flex-col truncate">
+          <div className="flex space-y-1">
+            <span
+              className={classNames(
+                "truncate text-sm",
+                props.webhook.active ? "text-neutral-700" : "text-neutral-200"
+              )}>
+              {props.webhook.subscriberUrl}
+            </span>
+          </div>
+          <div className="mt-2 flex">
+            <span className="flex flex-col space-x-2 space-y-1 text-xs sm:flex-row sm:space-y-0 sm:rtl:space-x-reverse">
+              {props.webhook.eventTriggers.map((eventTrigger, ind) => (
+                <span
+                  key={ind}
+                  className={classNames(
+                    "w-max rounded-sm px-1 text-xs ",
+                    props.webhook.active ? "bg-blue-100 text-blue-700" : "bg-blue-50 text-blue-200"
+                  )}>
+                  {t(`${eventTrigger.toLowerCase()}`)}
+                </span>
+              ))}
+            </span>
+          </div>
+        </div>
+        <div className="flex">
+          <Tooltip content={t("edit_webhook")}>
+            <Button
+              onClick={() => props.onEditWebhook()}
+              color="minimal"
+              size="icon"
+              StartIcon={PencilAltIcon}
+              className="ml-4 w-full self-center p-2"></Button>
+          </Tooltip>
+          <Dialog>
+            <Tooltip content={t("delete_webhook")}>
+              <DialogTrigger asChild>
+                <Button
+                  onClick={(e) => {
+                    e.stopPropagation();
+                  }}
+                  color="minimal"
+                  size="icon"
+                  StartIcon={TrashIcon}
+                  className="ml-2 w-full self-center p-2"></Button>
+              </DialogTrigger>
+            </Tooltip>
+            <ConfirmationDialogContent
+              variety="danger"
+              title={t("delete_webhook")}
+              confirmBtnText={t("confirm_delete_webhook")}
+              cancelBtnText={t("cancel")}
+              onConfirm={() =>
+                deleteWebhook.mutate({ id: props.webhook.id, eventTypeId: props.webhook.eventTypeId })
+              }>
+              {t("delete_webhook_confirmation_message")}
+            </ConfirmationDialogContent>
+          </Dialog>
+        </div>
+      </div>
+    </ListItem>
+  );
+}
+
+function WebhookTestDisclosure() {
```

Comment:
With this we now have multiple `WebhookTestDisclosure` components. My suggestion would be to try and make the original component reusable, extract it and reuse it in both places. This comment also applies to all repetitive logic we are making with web hooks in general. Also, I think we can make a single CRUD endpoint for all web hooks (user, team and eventType) instead of multiple ones.

![image](https://user-images.githubusercontent.com/3504472/155195642-a110b049-3e9b-4335-8947-19cfc1b88604.png)

Reply:
Yeah, I'm using the same CRUD endpoint for all webhooks, but I agree, we should extract the webhook disclosure components and all the repetitive webhook logic. Will add this.